### PR TITLE
Update to Jakarta messaging Client and other dependencies.

### PR DESCRIPTION
**Pull request by ksalazar-91**
Dependencies update to ensure security compliance and to move to jakarta for spring compliance. 

### Updating to newest version
- maven-plugin-plugin 3.11.0
- qpid-jms-client 2.5.0
- azure-identity 1.11.2

### Add
- junit 5.10.2
- jakarta.jms-api 2.0.3

### Remove
- geronimo-jms_2.0_spec
- junit 4

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -66,23 +66,29 @@
...
 		<dependency>
 		    <groupId>com.azure</groupId>
 		    <artifactId>azure-identity</artifactId>
-		    <version>1.7.3</version>
+		    <version>1.11.2</version>
+		</dependency>
+		<dependency>
+		    <groupId>org.slf4j</groupId>
+		    <artifactId>slf4j-api</artifactId>
+		    <version>2.1.0-alpha1</version>
```
Comment (vinaysurya) [inline pom.xml:85]: >2.1.0-alpha1 [](http://example.com/codeflow?start=15&length=12)

why move to an alpha package, can we just do the previous stable released version?

File: pom.xml [hunk 2]
```diff
@@ -66,23 +66,29 @@
...
+		</dependency>
+		<dependency>
+		    <groupId>org.slf4j</groupId>
+		    <artifactId>slf4j-simple</artifactId>
+		    <version>2.1.0-alpha1</version>
```
Comment (vinaysurya) [inline pom.xml:90]: >2.1.0-alpha1 [](http://example.com/codeflow?start=15&length=12)

same here, can we move to a released version?

**Review by vinaysurya — APPROVED** [review]

File: src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnection.java [hunk 3]
```diff
@@ -3,15 +3,15 @@
...
-import javax.jms.Topic;
+import jakarta.jms.Connection;
+import jakarta.jms.ConnectionConsumer;
+import jakarta.jms.ConnectionMetaData;
+import jakarta.jms.Destination;
+import jakarta.jms.ExceptionListener;
+import jakarta.jms.JMSException;
+import jakarta.jms.ServerSessionPool;
+import jakarta.jms.Session;
+import jakarta.jms.Topic;
```
Comment (vinaysurya) [inline src/main/java/com/microsoft/azure/servicebus/jms/ServiceBusJmsConnection.java:14]: Just adding a comment here but that is really for the PR description:

1.  Remove this in PR description:

During testing I found out Jndi bug introduced by autodelete on entities

2. For title, lets use something like:
Update to Jakarta messaging Client and other dependencies.

File: pom.xml (hunk 1 above)
Comment (ksalazar-91) [reply pom.xml]: No reason just testing, I forgot to change to a more stable.

File: pom.xml (hunk 2 above)
Comment (ksalazar-91) [reply pom.xml]: done

File: pom.xml [hunk 4]
```diff
@@ -4,7 +4,7 @@
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>com.microsoft.azure</groupId>
 	<artifactId>azure-servicebus-jms</artifactId>
-	<version>1.0.1</version>
+	<version>2.0.0-beta.0</version>
```
Comment (vinaysurya) [inline pom.xml:7]: >2.0.0-beta.0< [](http://example.com/codeflow?start=10&length=13)

the convention that sdk team seems for beta versions is to got with a minor version 1. 

[https://mvnrepository.com/artifact/com.azure/azure-messaging-servicebus](https://mvnrepository.com/artifact/com.azure/azure-messaging-servicebus)

So lets also do the same and make it 2.0.0-beta.1

**Review by vinaysurya — APPROVED** [review]

**Comment by tngafwlehmann** [comment]
Thanks for working on this! May I ask if there's a release planned soon?

**Comment by tngafwlehmann** [comment]
> Thanks for working on this! May I ask if there's a release planned soon?

Nevermind. I just realized that it's out :)

**Pull request merged**